host/frontend/host-orchestrator/instancemanager.go

```diff
@@ -14,12 +14,36 @@
 
 package main
 
-type InstanceManager struct{}
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"sync"
+)
+
+type InstanceManager struct {
+	fetchCVDHandler *FetchCVDHandler
+}
+
+func NewInstanceManager(fetchCVDHandler *FetchCVDHandler) *InstanceManager {
+	return &InstanceManager{fetchCVDHandler}
+}
 
 func (m *InstanceManager) CreateCVD(req *CreateCVDRequest) (*Operation, error) {
 	if err := validateRequest(req); err != nil {
 		return nil, err
 	}
+	// This logic isn't complete yet, it's work in progress.
+	go func() {
+		err := m.fetchCVDHandler.Download(req.FetchCVDBuildID, req.BuildAPIAccessToken)
```

Comment:
Is the cloud orchestrator handing out tokens here to host orchestrators to access the build API? Is the cloud orchestrator safe against rogue clients claiming to be host orchestrators?

In the specific case of getting `fetch_cvd` exectuables off `aosp-master` builds on [ci.android.com](https://ci.android.com), that should be possible without credentials. In the case of giving `fetch_cvd` credentials to download the rest of the images, `fetch_cvd` is capable of pulling credentials from the GCE environment, which seems relevant to the cloud GCE case.

I can see the case for only-premises specifically, but the security issue also stands out stronger there depending on how instance acquisition there works.

Reply:
> Is the CO safe agains fake HO?
That's a very good question and the answer is "not today". I'm planning to add some hardening to this communications so that the CO and HO can authenticate each other, so that unless the rogue client would need to be able to become root in the host to be able to impersonate the HO.